src/heretic/model.py

```diff
@@ -233,21 +276,54 @@ def abliterate(
                 # W_new = W - α(r (r^T W))
                 r = layer_refusal_direction.to(self.model.dtype)
 
-                for matrix in matrices:
-                    # Ensure r is on the same device as the matrix for multi-GPU support.
-                    r_device = r.to(matrix.device)
-
-                    # Calculate the projection scalars: (r^T W)
-                    # r is (d,), matrix is (d, k) -> result is (k,)
-                    r_transpose_W = torch.matmul(r_device, matrix)
-
-                    # Compute the rank-1 update r (r^T W) using the outer product form
-                    # r_device: (d,)          — projection direction
-                    # r_transpose_W: (k,)     — r^T W result for this matrix
-                    # torch.outer(r_device, r_times_W) constructs the (d, k) matrix with
-                    # entries r[i] * (r^T W)[j], equivalent to the outer product of two
-                    # vectors, avoiding materializing the full (d x d) projector.
-                    matrix.sub_(weight * torch.outer(r_device, r_transpose_W))
+
+                for module in modules:
+                    if self.settings.use_lora and hasattr(module, "lora_A"):
```

Comment:
So the `hasattr` check handles the case where the "module" is actually a tensor, correct?